source/plugins/go/src/oms.go

```diff
@@ -1763,17 +1831,46 @@ func InitializePlugin(pluginConfPath string, agentVersion string) {
 			Log("Routing container logs thru %s route...", ContainerLogsADXRoute)
 			fmt.Fprintf(os.Stdout, "Routing container logs thru %s route...\n", ContainerLogsADXRoute)
 		}
-	} else if strings.Compare(strings.ToLower(osType), "windows") != 0 { //for linux, oneagent will be default route
+	} else if IsWindows == false { //for linux, oneagent will be default route
 		ContainerLogsRouteV2 = true //default is mdsd route
 		Log("Routing container logs thru %s route...", ContainerLogsRoute)
 		fmt.Fprintf(os.Stdout, "Routing container logs thru %s route... \n", ContainerLogsRoute)
+	} else if IsAADMSIAuthMode || IsGenevaLogsIntegrationEnabled { //for windows, check if MSI Auth mode, then send through AMA
+		ContainerLogsRouteV2 = true
+		Log("Routing container logs thru %s route...", ContainerLogsV2Route)
+		fmt.Fprintf(os.Stdout, "Routing container logs thru %s route... \n", ContainerLogsV2Route)
+	}
+
+	EnvContainerLogSchemaVersion := strings.TrimSpace(strings.ToLower(os.Getenv("AZMON_CONTAINER_LOG_SCHEMA_VERSION")))
+	Log("AZMON_CONTAINER_LOG_SCHEMA_VERSION:%s", EnvContainerLogSchemaVersion)
+
+	ContainerLogSchemaV2 = false //default is v1 schema
+
+	if strings.Compare(EnvContainerLogSchemaVersion, ContainerLogV2SchemaVersion) == 0 && ContainerLogsRouteADX != true {
+		ContainerLogSchemaV2 = true
+		Log("Container logs schema=%s", ContainerLogV2SchemaVersion)
+		fmt.Fprintf(os.Stdout, "Container logs schema=%s... \n", ContainerLogV2SchemaVersion)
 	}
 
 	if ContainerLogsRouteV2 == true {
-		CreateMDSDClient(ContainerLogV2, ContainerType)
+		if IsWindows {
+			var datatype string
+			if ContainerLogSchemaV2 {
+				datatype = ContainerLogV2DataType
+			} else {
+				datatype = ContainerLogDataType
+			}
+			if IsGenevaLogsIntegrationEnabled {
+				CreateWindowsNamedPipesClient(getGenevaWindowsNamedPipeName())
+			} else {
+				CreateWindowsNamedPipesClient(extension.GetInstance(FLBLogger, ContainerType).GetOutputNamedPipe(datatype))
+			}
+		} else {
+			CreateMDSDClient(ContainerLogV2, ContainerType)
+		}
 	} else if ContainerLogsRouteADX == true {
 		CreateADXClient()
-	} else { // v1 or windows
+	} else { // windows without MSI AUTH
 		Log("Creating HTTP Client since either OS Platform is Windows or configmap configured with fallback option for ODS direct")
 		CreateHTTPClient()
 	}
```

Comment:
Keep the whole oms.go as it is and only change where Geneva integration is needed. We havent merged Container Log yet with Windows AMA

Reply:
> Keep the whole oms.go as it is and only change where Geneva integration is needed. We havent merged Container Log yet with Windows AMA

Done.